src/status_im/chat/events/input.cljs

```diff
@@ -0,0 +1,496 @@
+                        (validator params error-events-creator))]
+     {:dispatch-n (or error-events proceed-events)})))
+
+(register-handler-fx
+ ::send-command
+ [trim-v]
+ (fn [_ [on-send {{:keys [fullscreen bot]} :command :as content} chat-id]]
+   (if on-send
+     {:dispatch-n (cond-> [[:set-chat-ui-props {:result-box           on-send
+                                                :sending-in-progress? false}]]
+                    fullscreen
+                    (conj [:choose-predefined-expandable-height :result-box :max]))
+      ::dismiss-keyboard nil}
+     {:dispatch [::request-command-data
+                 {:content         content
+                  :chat-id         chat-id
+                  :jail-id         (or bot chat-id)
+                  :data-type       :preview
+                  :event-after     (fn [command-message _]
+                                     [::send-message command-message chat-id])}]})))
+
+(register-handler-fx
+ ::request-command-data
+ [trim-v (inject-cofx :random-id) (inject-cofx :now)]
+ (fn [{{:keys [bot-db] :contacts/keys [contacts]} :db
+       message-id :random-id
+       current-time :now}
+      [{{:keys [command
+                metadata
+                args]
+         :as   content} :content
+        :keys  [chat-id jail-id data-type event-after]}]]
+   (let [{:keys [dapp? dapp-url name]} (get contacts chat-id)
+         metadata        (merge metadata
+                                (when dapp?
+                                  {:url  (i18n/get-contact-translated chat-id :dapp-url dapp-url)
+                                   :name (i18n/get-contact-translated chat-id :name name)}))
+         owner-id        (:owner-id command)
+         bot-db          (get bot-db chat-id)
+         params          (merge (input-model/args->params content)
+                                {:bot-db   bot-db
+                                 :metadata metadata})
+
+         command-message {:command    command
+                          :params     params
+                          :to-message (:to-message-id metadata)
+                          :created-at current-time
+                          :id         message-id
+                          :chat-id    chat-id
+                          :jail-id    (or owner-id jail-id)}
+
+         request-data    {:message-id   message-id
+                          :chat-id      chat-id
+                          :jail-id      (or owner-id jail-id)
+                          :content      {:command (:name command)
+                                         :params  params
+                                         :type    (:type command)}
+                          :on-requested (fn [jail-response]
+                                          (event-after command-message jail-response))}]
+     {:dispatch [:request-command-data request-data data-type]})))
+
+(register-handler-fx
+ :send-current-message
```

Comment:
ns keyword (there is more below I won't repeat my comment on each of them you get the point)